clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -217,7 +220,7 @@ protected CompletableFuture<V> get(GetRequestContext requestContext, K key) thro
      * might return more than required number of routes
      */
     List<String> routes = metadata.getReplicas(
-        requestContext.requestId,
+        REQUEST_ID_GENERATOR.getAndIncrement(),
```

Comment:
can you elaborate more on why this change is helpful?

Reply:
Currently, in `RetriableAvroGenericStoreClient`, when retry is enabled, for multi-get request, it will generate 3 RequestContexts: original one passed by the upstreaming layer, a new one for the original request and a new one for the retry request, so request-id is kind of messy-up even it would only send out two requests at most.
The request id generation in `DispatchingAvroGenericStoreClient.java` will guarantee the request id would only be bumped when there is a request to be sent out, and this also decouples request id generation from the `RequestContext` creation to be less error prone.